tools/DfmHttpService/DfmHttpService.cs

```diff
@@ -0,0 +1,168 @@
+
+        public void StartService(string urlPrefix)
+        {
+            var listener = new HttpListener();
+            listener.Prefixes.Add(urlPrefix);
+            listener.Start();
+
+            while (KeepGoing)
+            {
+                var context = listener.GetContext();
+                HandleRequest(context);
+            }
+
+            listener.Stop();
+            listener.Close();
+        }
+
+        private void HandleRequest(HttpListenerContext context)
+        {
+            // TODO: Add log for request information
+            var request = context.Request;
+
+            CommandMessage command;
+            try
+            {
+                command = GetCommandMessage(request);
+            }
+            catch (Exception ex)
+            {
+                ReplyClientErrorResponse(context, ex.Message);
+                return;
+            }
+
+            switch (command.Name)
+            {
+                case "preview":
+                    string content;
+                    try
+                    {
+                        content = Preview(command.Documentation, command.FilePath, command.WorkspacePath);
+                    }
+                    catch (Exception ex)
+                    {
+                        ReplyServerErrorResponse(context, ex.Message);
+                        return;
+                    }
+                    ReplySuccessfulResponse(context, content);
+                    return;
+                case "generateTokenTree":
+                    string tokenTree;
+                    try
+                    {
+                        tokenTree = GenerateTokenTree(command.Documentation, command.FilePath, command.WorkspacePath);
+                    }
+                    catch (Exception ex)
+                    {
+                        ReplyServerErrorResponse(context, ex.Message);
+                        return;
+                    }
+                    ReplySuccessfulResponse(context, tokenTree);
+                    return;
+                case "exit":
+                    KeepGoing = false;
+                    ReplyExitResponse(context, "Dfm service exit");
+                    return;
+                default:
+                    ReplyClientErrorResponse(context, "Can't find the name of command");
+                    return;
+            }
+        }
+
+        public static string GetAvailablePrefix()
+        {
+            // TODO: generate new random port if default port is not avaliable
+            return UrlPrefix;
+        }
+
+        private static string GenerateTokenTree(string documentation, string filePath, string workspacePath = null)
+        {
+            var provider = new DfmJsonTokenTreeServiceProvider();
+            var service = provider.CreateMarkdownService(new MarkdownServiceParameters { BasePath = workspacePath});
+
+            return service.Markup(documentation, filePath).Html;
```

Comment:
I think it is a Json string which should be renamed.

---
In reply to: [94094956](https://github.com/dotnet/docfx/pull/1075#discussion_r94094956) [](ancestors = 94094956,94094589)